experiments/signals.py

```diff
@@ -17,3 +17,38 @@ def on_datapoint_creation(sender, instance, *args, **kwargs):
             instance.number = last_datapoint.number + 1
         else:
             instance.number = 1
+
+
+@receiver(pre_save, sender=ParticipantSession)
+def on_participant_session_creation(
+        sender,
+        instance: ParticipantSession,
+        *args,
+        **kwargs):
+    """
+    Add an incremental session_id number when saving a session starting from 1
+    """
+    if not instance.subject_id:
+        experiment = instance.experiment
+        last_session = experiment.participantsession_set.all().order_by(
+            'subject_id'
+        ).last()
+
+        if last_session:
+            instance.subject_id = last_session.subject_id + 1
+        else:
+            instance.subject_id = 1
+
+
+@receiver(post_save, sender=Experiment)
+def on_experiment_creation(
+        sender,
+        instance: Experiment,
+        *args,
+        **kwargs):
+    """Add one default group, with "list1" as name"""
+    if instance.targetgroup_set.count() == 0:
+        instance.targetgroup_set.create(
+            name="list1",
```

Comment:
I have chosen "list1" with purpose. In the Zep/jsPsych experiments the participants get assigned a list of stimuli. In most of our boilerplate experiments we have 2 lists, the experimental items are interleaved with _fillers_, in the other list, the fillers are experimental stimuli and vice versa. This a/the way of implementing a latin square design. So I've chozen this name to match the listname e.g. in:
https://github.com/UiL-OTS-labs/jspsych-spr-mw/blob/main/stimuli.js line 15 and 16.
Because eventually, this is why we wanted to have "grouping/listing" in the first place.

Reply:
Hm, I see. I think this discussion will boil down to 'how generic do we want this app to be'; but honestly, I'm not really in the mood for this discussion ;)
Could we maybe compromise on 'List 1' instead of 'list1'? It's user interface text, so I'd rather have something more human than code, and I think this would still be in line enough with the boilerplates

Reply:
I'll just update it to "Default Group" then, as we need to map the pretty group name to a useful one in the javascript code anyway in this case.

Reply:
fixed in https://github.com/UiL-OTS-labs/web-experiment-datastore/pull/39/commits/70ff99513ba0226d896618d28b07db425bff5287